File: web/src/plugin/mock_subscribe/api/merchant.js
import service from '@/utils/request'

export const createMerchant = (data) => {
  return service({
    url: '/mockSubscribeMerchant/createMerchant',
    method: 'post',
    data
  })
}

export const updateMerchant = (data) => {
  return service({
    url: '/mockSubscribeMerchant/updateMerchant',
    method: 'put',
    data
  })
}

export const deleteMerchant = (params) => {
  return service({
    url: '/mockSubscribeMerchant/deleteMerchant',
    method: 'delete',
    params
  })
}

export const findMerchant = (params) => {
  return service({
    url: '/mockSubscribeMerchant/findMerchant',
    method: 'get',
    params
  })
}

export const getMerchantList = (params) => {
  return service({
    url: '/mockSubscribeMerchant/getMerchantList',
    method: 'get',
    params
  })
}

File: server/plugin/mock_subscribe/router/merchant.go
package router

import (
	"github.com/flipped-aurora/gin-vue-admin/server/middleware"
	"github.com/gin-gonic/gin"
)

type merchant struct{}

func (r *merchant) Init(public *gin.RouterGroup, private *gin.RouterGroup) {
	_ = public
	group := private.Group("mockSubscribeMerchant").Use(middleware.OperationRecordWithUserID(3))
	group.POST("createMerchant", apiInfo.Merchant.CreateMerchant)
	group.PUT("updateMerchant", apiInfo.Merchant.UpdateMerchant)
	group.DELETE("deleteMerchant", apiInfo.Merchant.DeleteMerchant)
	group.GET("findMerchant", apiInfo.Merchant.FindMerchant)
	group.GET("getMerchantList", apiInfo.Merchant.GetMerchantList)
}
